File: presentation/schemas/assignment.ts
export interface NewAssignmentSchema {
  old_interface_id: number;
  current_interface_id: number;
  username: string;
  assign_by: string;
  type_status: string;
}

export interface ReassignmentSchema {
  old_interface_id: number;
  current_interface_id: number;
  old_username: string;
  new_username: string;
  assign_by: string;
}

export interface UpdateAssignmentSchema {
  old_interface_id: number;
  current_interface_id: number;
  type_status: string;
}

export interface StatisticsAssignmentSchema {
  total_pending_today: number;
  total_inspected_today: number;
  total_rediscovered_today: number;
  total_pending_month: number;
  total_inspected_month: number;
  total_rediscovered_month: number;
  username: string;
  name: string;
  lastname: string;
}

File: icm/access/models/assignment.py
from pydantic import BaseModel


class ReassignmentModel(BaseModel):
    old_interface_id: int
    current_interface_id: int
    old_username: str
    new_username: str
    assign_by: str


class UpdateAssignmentModel(BaseModel):
    old_interface_id: int
    current_interface_id: int
    username: str
    type_status: str


class StatisticsModel(BaseModel):
    total_pending_today: int
    total_inspected_today: int
    total_rediscovered_today: int
    total_pending_month: int
    total_inspected_month: int
    total_rediscovered_month: int
    username: str
    name: str
    lastname: str